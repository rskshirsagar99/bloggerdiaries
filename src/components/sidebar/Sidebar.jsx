import axios from "axios";
import { useEffect, useState } from "react";
import { Link } from "react-router-dom";
import "./sidebar.css";


export default function Sidebar() {
  var username = sessionStorage.getItem("username")
 
  return (
    <div className="sidebar">
      <div className="sidebarItem">
        <div className="heading">
          <h2>Read, Write and Expand Your World.</h2>
        </div>
        <div className="animation">
        <iframe src="https://player.vimeo.com/video/467834328?api=1&amp;background=1&amp;mute=1&amp;loop=1" width="100%" height="100%" positioning="relative" frameborder="0" allow="autoplay; fullscreen" allowfullscreen=""></iframe>
        </div>
      </div>
      
      <div className="sidebarItem">
        <span className="sidebarTitle">CATEGORIES</span>
        <ul className="sidebarList">
          <li className="sidebarListItem"><Link to="?cat=Health">Health</Link></li>
          <li className="sidebarListItem"><Link to="?cat=Sports">Sports</Link></li>
          <li className="sidebarListItem"><Link to="?cat=Music">Music</Link></li>
          <li className="sidebarListItem"><Link to="?cat=Food">Food</Link></li>
          <li className="sidebarListItem"><Link to="?cat=Technology">Technology</Link></li>
          <li className="sidebarListItem"><Link to="?cat=Others">Others</Link></li>
          <li className="sidebarListItem"><Link to={`?user=${username}`}>My Blogs</Link></li>
          
        </ul>
      </div>
    </div>
  );
}
