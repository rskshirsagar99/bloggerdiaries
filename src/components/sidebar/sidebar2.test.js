import React from 'react'
import renderer from 'react-test-renderer'
import Sidebar from './Sidebar';

import { MemoryRouter } from 'react-router-dom';




test('snapshot test of Sidebar',()=>{

  const tree=renderer
  .create(
    <MemoryRouter>
      <Sidebar/>
    </MemoryRouter>
  ).toJSON()
  expect(tree).toMatchSnapshot()


})