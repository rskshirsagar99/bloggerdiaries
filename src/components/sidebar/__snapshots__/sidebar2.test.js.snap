// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test of Sidebar 1`] = `
<div
  className="sidebar"
>
  <div
    className="sidebarItem"
  >
    <div
      className="heading"
    >
      <h2>
        Read, Write and Expand Your World.
      </h2>
    </div>
    <div
      className="animation"
    >
      <iframe
        allow="autoplay; fullscreen"
        allowfullscreen=""
        frameborder="0"
        height="100%"
        positioning="relative"
        src="https://player.vimeo.com/video/467834328?api=1&background=1&mute=1&loop=1"
        width="100%"
      />
    </div>
  </div>
  <div
    className="sidebarItem"
  >
    <span
      className="sidebarTitle"
    >
      CATEGORIES
    </span>
    <ul
      className="sidebarList"
    >
      <li
        className="sidebarListItem"
      >
        <a
          href="/?cat=Health"
          onClick={[Function]}
        >
          Health
        </a>
      </li>
      <li
        className="sidebarListItem"
      >
        <a
          href="/?cat=Sports"
          onClick={[Function]}
        >
          Sports
        </a>
      </li>
      <li
        className="sidebarListItem"
      >
        <a
          href="/?cat=Music"
          onClick={[Function]}
        >
          Music
        </a>
      </li>
      <li
        className="sidebarListItem"
      >
        <a
          href="/?cat=Food"
          onClick={[Function]}
        >
          Food
        </a>
      </li>
      <li
        className="sidebarListItem"
      >
        <a
          href="/?cat=Technology"
          onClick={[Function]}
        >
          Technology
        </a>
      </li>
      <li
        className="sidebarListItem"
      >
        <a
          href="/?cat=Others"
          onClick={[Function]}
        >
          Others
        </a>
      </li>
    </ul>
  </div>
</div>
`;
